src/plugins/segmentio/__tests__/index.test.ts

```diff
@@ -105,17 +112,22 @@ describe('Segment.io', () => {
 
   describe('#track', () => {
     it('should enqueue an event and properties', async () => {
```

Comment:
This test appears to be passing even without your change. I think that's because your stubbed class doesn't throw any errors in the constructor - so even the current implementation will be able to instantiate the class and then copy the properties over.

In the example I saw, the instantiation of the new class would fail due to an error (I believe a `TypeError` due to trying to access a field on an `undefined` object.) For testing, you can mimic that by throwing an error in the constructor if the parameters aren't passed.

Reply:
We had suggested to the client to clone their class instance into a new object, because it appeared that using a class instance did not play well with the data flow.  It sounds like the problem may be elsewhere if a unit test trying to mimic this experience ends up green. I'll take a closer look at things on my end.

Reply:
This was a good catch, @chrisradek. I do think there's also an element of something you mentioned here going wrong, in tandem with the way that the data being passed as a class is incongruent with the cloning utility.

Reply:
In one user's case, their class was something like:
```js
class Foo {
  constructor(props) {
    Bar.call(this, props.a, props.b, props.c);
  }
}
```
Since the cloning utility doesn't have access to `props` to pass in when instantiating the class, the constructor threw a TypeError since it was trying to access fields on `props` while `props` was undefined. I think copying the fields to a new plain object would have worked fine (like your change does).

Anyway, that's why I think we could reproduce the issue in a test easily enough by just throwing a TypeError if nothing is passed to the constructor.

Reply:
Yep, that's what I noticed as well. I'll get a test up and running to reflect the issue and the fix better.